<?xml version="1.0"?>
<launch>
    <!-- ARGUMENTS - BRINGUP -->
    <arg name="mode" default="3d" doc="modes [2d, 3d]"/>
    
    <!-- ARGUMENTS - OAKD -->
    <arg name="camera_name" default="oakd"/>
    <arg name="oakd_mode"   default="rgbd" doc="OAK-D Modes [rgbd, stereo (only for testing/debugging)]"/>

    <!-- INCLUDE DESCRIPTION LAUNCH FILE -->
    <include file="$(find akros_test)/launch/oakd_description.launch"/>
    
    <!-- IF 3D, LAUNCH OAK-D -->
    <group if="$(eval arg('mode') == '3d')"> 
        <!-- INCLUDE OAK-D LAUNCH FILE -->
        <include file="$(find akros_test)/launch/demo_oakd.launch">
            <arg name="camera_name" value="$(arg camera_name)"/>
            <arg name="oakd_mode" value="$(arg oakd_mode)"/>
            <arg name="standalone" value="false"/>
        </include>
    </group>
    
    <!-- IF NOT 3D, LAUNCH OAK-D IMU -->
    <group unless="$(eval arg('mode') == '3d')">
        <!-- INCLUDE OAK-D IMU LAUNCH FILE -->
        <!-- launch-prefix="xterm -e gdb (add [- - args] without space) -->
        <node name="oakd" pkg="akros_bringup" type="oakd_imu_node" output="screen" required="true">
            <param name="camera_name" value="$(arg camera_name)"/>
        </node>
    </group>

</launch>
